codepropertygraph/src/main/scala/io/shiftleft/passes/ParallelCpgPass.scala

```diff
@@ -38,6 +38,10 @@ abstract class ParallelCpgPass[T](cpg: Cpg, outName: String = "", keyPools: Opti
     val writer = new Writer(serializedCpg, prefix, inverse, MDC.getCopyOfContextMap())
     val writerThread = new Thread(writer)
     writerThread.setName("Writer")
+    var exceptionCaught: Option[Throwable] = None
```

Comment:
This is ok because `writerThread.join()` is a memory barrier; `@volatile` would come at no perceptible performance cost and make it "more obviously OK".

Feel free to disregard the suggestion if you disagree, that's a pure coding style question.